README.md

```diff
@@ -1,3 +1,317 @@
+
+Some nice things about `mbq.apitools`:
+* All fields specified in a param or payload schema are required by default, and can be marked as optional by providing a `default=` argument to the field class. The framework will automatically return a 400 response for all requests which do not conform to the specified schema. Details for each nonconforming field will be included in the response.
+* The parsed parameters and payloads end up as rich types on the request. If a field is marked as `fields.DateTime`, then it will be on `request.payload` as a `datetime` object.
+* Pagination is handled entirely by the framework. Just mark a view as paginated and return an instance of a `PaginatedResponse` and voila!
+*  All success responses have a 200 status code.
+* All list and paginated responses contain the list of resources under the key `"objects"`.
+* All error responses (4xx) have the same shape (an `"error_code"` and `"detail"` key).
+
+## Example
+
+```python
+from mbq.api_tools import fields, responses
+from mbq.api_tools.views import View, view
+
+@view(
+    "GET",
+    permissions=[SomeDRFPermissionClass],
+    params={
+        "product_ids": fields.Int(default=None, many=True),
+        "zipcode": fields.String(),
+    }
+)
+def get_categories(request):
+    categories = Category.objects.filter(zipcode=request.params.zipcode)
+
+    if request.product_ids:
+        categories = categories.filter(product_id__in=request.params.product_ids)
+
+    categories = CategorySerializer(categories, many=True).data
+
+    return responses.ListResponse(categories)
+
+
+class OrdersView(View):
+
+    @view.method("POST", payload={"company_id": fields.Int()})
+    def create_order(self, request):
+        order = create_order(request.payload.company_id)
+        return responses.DetailResponse(OrderSerializer(order).data))
+
+    @view.method("GET", permissions=[SomeDRFPermission], paginated=True)
+    def get_orders(self, request):
+        orders = Order.objects.all()
+        return responses.PaginatedResponse(orders, OrderSerializer, request)
+```
+## Views
+
+### @view
+
+Use the `@view` decorator for all function based views. It accepts the following arguments:
+* `http_method_name`
+	* GET, POST, PATCH, PUT, DELETE
+	* Defaults to GET
+* `permissions`
+	* A list of DRF permission classes
+		* They don't necessarily need to subclass a DRF permission class, they only need to have a `has_permission(request, view)` method.
+* `params`
+	* Schema for validating incoming query parameters. The query parameters will be available at `request.params`.
+* `payload`
+	* Schema for validating incoming payloads. The payload will be available at `request.payload`.
+* `paginated`
+	* Boolean indicating if the response will be paginated or not. Defaults to `False`.
+* `page_size`
+	* Integer specifying the page size for a paginated response. This should only be used to override the default page size.
```

Comment:
What is the default page size?

Reply:
20 🙃 
https://github.com/managedbyq/mbq.apitools/pull/8/files#diff-04c6e90faac2675aa89e2176d2eec7d8R286